crates/engine/tree/src/tree/mod.rs

```diff
@@ -2161,7 +2161,8 @@ where
         //
         // See https://github.com/paradigmxyz/reth/issues/12688 for more details
         let persisting_kind = self.persisting_kind_for(block.header());
-        let run_parallel_state_root = persisting_kind.can_run_parallel_state_root();
+        let run_parallel_state_root = persisting_kind.can_run_parallel_state_root() &&
+            !self.config.test_state_root_fallback(); // don't run parallel if testing fallback
```

Comment:
nit: would prefer the comment to be just before the `let`

Reply:
yep, now it also mentions it is fallback for state root